### ISSUE TITLE
Contextive LSP fails with unhandled exception

### ISSUE THREAD

**Issue by ssiltanen** [dev-cycles/contextive#60]
**Describe the bug**
First time opening Contextive LSP with Helix editor causes Contextive LSP to throw an unhandled exception. The unhandled exception is `Method not found: window/showMessageRequest` which seems to be coming from the survey code.

**To Reproduce**
1. Install helix https://helix-editor.com/
2. Install Contextive 1.10.3
3. Configure helix with contextive for typescript by adding `~/.config/helix/languages.toml` config with
```
[[language]]
name = "typescript"
language-servers = [{ name = "contextive" }]

[language-server.contextive]
args = [ "--stdio"]
command = "Contextive.LanguageServer"
```
4.  Add example contextive config to your repo root: `.contextive/definitions.yml` from https://github.com/dev-cycles/contextive-demo-go-common/blob/main/.contextive/definitions.yml
5. Hover "Sample" anywhere in your repo by pressing `<space>+k` when cursor is on top of the word "Sample"

**Expected behavior**
Contextive should give tooltip about the term

**Helix log of the error**
```
2023-11-17T17:44:49.725 helix_term::application [ERROR] Language Server: Method window/showMessageRequest not found in request 1
2023-11-17T17:44:49.767 helix_lsp::transport [ERROR] contextive err <- "Unhandled exception. System.AggregateException: One or more errors occurred. (Method not found: 'window/showMessageRequest'.)\n"
2023-11-17T17:44:49.767 helix_lsp::transport [ERROR] contextive err <- " ---> OmniSharp.Extensions.JsonRpc.Server.MethodNotSupportedException: Method not found: 'window/showMessageRequest'.\n"
2023-11-17T17:44:49.767 helix_lsp::transport [ERROR] contextive err <- "   at OmniSharp.Extensions.JsonRpc.ResponseRouter.ResponseRouterReturnsImpl.Returning[TResponse](CancellationToken cancellationToken)\n"
2023-11-17T17:44:49.767 helix_lsp::transport [ERROR] contextive err <- "   at Contextive.LanguageServer.Survey.showSurveyPrompt@11.MoveNext() in /Users/runner/work/contextive/contextive/src/language-server/Contextive.LanguageServer/Survey.fs:line 31\n"
2023-11-17T17:44:49.767 helix_lsp::transport [ERROR] contextive err <- "   at OmniSharp.Extensions.LanguageServer.Server.LanguageServer.Initialize(CancellationToken token)\n"
2023-11-17T17:44:49.767 helix_lsp::transport [ERROR] contextive err <- "   at OmniSharp.Extensions.LanguageServer.Server.LanguageServer.From(LanguageServerOptions options, IServiceProvider outerServiceProvider, CancellationToken cancellationToken)\n"
2023-11-17T17:44:49.767 helix_lsp::transport [ERROR] contextive err <- "   --- End of inner exception stack trace ---\n"
2023-11-17T17:44:49.767 helix_lsp::transport [ERROR] contextive err <- "   at Microsoft.FSharp.Control.AsyncResult`1.Commit() in D:\\a\\_work\\1\\s\\src\\FSharp.Core\\async.fs:line 454\n"
2023-11-17T17:44:49.767 helix_lsp::transport [ERROR] contextive err <- "   at Microsoft.FSharp.Control.AsyncPrimitives.QueueAsyncAndWaitForResultSynchronously[a](CancellationToken token, FSharpAsync`1 computation, FSharpOption`1 timeout) in D:\\a\\_work\\1\\s\\src\\FSharp.Core\\async.fs:line 1137\n"
2023-11-17T17:44:49.767 helix_lsp::transport [ERROR] contextive err <- "   at Microsoft.FSharp.Control.AsyncPrimitives.RunSynchronously[T](CancellationToken cancellationToken, FSharpAsync`1 computation, FSharpOption`1 timeout) in D:\\a\\_work\\1\\s\\src\\FSharp.Core\\async.fs:line 1167\n"
2023-11-17T17:44:49.767 helix_lsp::transport [ERROR] contextive err <- "   at Microsoft.FSharp.Control.FSharpAsync.RunSynchronously[T](FSharpAsync`1 computation, FSharpOption`1 timeout, FSharpOption`1 cancellationToken) in D:\\a\\_work\\1\\s\\src\\FSharp.Core\\async.fs:line 1509\n"
2023-11-17T17:44:49.767 helix_lsp::transport [ERROR] contextive err <- "   at Contextive.LanguageServer.Program.main(String[] argv) in /Users/runner/work/contextive/contextive/src/language-server/Contextive.LanguageServer/Program.fs:line 36\n"
2023-11-17T17:44:49.768 helix_lsp::transport [ERROR] contextive err: <- StreamClosed
2023-11-17T17:44:50.361 helix_term::application [ERROR] received malformed notification from Language Server: Unhandled
```

**Desktop:**
 - OS: MacOS Sonoma 14.1
 - Editor: Helix 23.10
 - .NET SDKs installed 6.0.417, 8.0.100
 - Contextive 1.10.3

**Comment by chrissimon-au** [comment]
Thanks for the report! I had assumed that the LSP server library I was using would silently drop the `window/showMessageRequest` if the client doesn't support it (it does that with some other operations, such as `workspace/configuration`, but it seems I need to explicitly check for support.

Should be a fairly quick fix!

Thanks also for the info about configuring helix - I'll borrow that for the Contextive readme documentation if that's ok?

**Comment by ssiltanen** [comment]
>Thanks also for the info about configuring helix - I'll borrow that for the Contextive readme documentation if that's ok?

Sure! I can report back once the LSP setup works if anything else was needed to set it up

**Comment by chrissimon-au** [comment]
Hi @ssiltanen [v1.10.4](https://github.com/dev-cycles/contextive/releases/tag/v1.10.4) should resolve this issue - let me know how you go!  As part of reproducing/testing, I've also added some documentation to the readme - let me know if this makes sense and works for you: https://github.com/dev-cycles/contextive#helix (note, it's basically yours, but you don't need the `--stdio` arg, and I've included what to do if the definitions file is not in the default location.

As a helix user, you might also be interested in this discussion: https://github.com/helix-editor/helix/discussions/8850 - about using an LSP on all files.  If they describe how to do that, or decide to update helix to support it, I'll update the documentation.

It also seems that I could open a PR against https://github.com/helix-editor/helix/blob/master/languages.toml to make the language-server configuration 'baked into' helix - it would then just require the user to nominate which languages to enable it for.  I'll wait until the discussion above is resolved before doing that, though, to ensure I understand the intended approach for an lsp like this.

Feel free to raise another issue if any problems!

**Comment by chrissimon-au** [comment]
Hold on that, seems to be an issue with 1.10.4.  Will be getting a fix out shortly.

**Comment by chrissimon-au** [comment]
OK [1.10.5](https://github.com/dev-cycles/contextive/releases/tag/v1.10.5) should work.

**Comment by ssiltanen** [comment]
> [v1.10.4](https://github.com/dev-cycles/contextive/releases/tag/v1.10.4) should resolve this issue

Yes, it resolves the issue! Thank you for work 💯 

> I've also added some documentation to the readme

Documentation looks correct to me and I can verify the `--stdio` was not needed after all like you mentioned.

> discussion about using an LSP on all files

Thank you, I was not aware of that discussion taking place yet, but that would be indeed cool to have, especially with this LSP. I agree before creating the helix PR, it is wise to follow the discussion you mentioned.

Also, this might be just a user error on my behalf but I noticed a behaviour with Helix and multi LSPs that the onHover shows docs only of one LSP, which affects Contextive as well. I started a discussion about it here https://github.com/helix-editor/helix/discussions/8872
